Formspecs: Add state-selection to style elements (#9378)
minetest/minetest: doc/lua_api.txt

        * `span=<value>`: number of following columns to affect
          (default: infinite).

> ### `style[<name 1>,<name 2>;<prop1>;<prop2>;...]`

Reviewer: how about:

```
### `style[<selector1>,<selector2>,...;<prop1>;<prop2>;...]`

* A <selector> can be one of:
    * `<name>` - an element name
    * `<name>:<state>` an element name, a colon, and an element state
```

### `style[<name 1>:<state>,<name 2>:state;<prop1>;<prop2>;...]`

* Set the style for the named element(s) `name`.

> * `state` is a list of states separated by the `+` character.

Reviewer: ```suggestion
* `<state>` is a list of states separated by the `+` character.
```

Author: Actually, @SmallJoker specifically asked me to change those to that format so that they better match the format of other formspec elements in the API.

>     * If a state that the element supports is provided, the style will only

Reviewer: ```suggestion
    * If a state is provided, the style will only take affect if the element
      is in that state. If an element doesn't support a state, it will never be
      in that state.
```

Reviewer 2: `If an element doesn't support a state, it will never be in that state.`
Ah yes, the floor is made out of floor.

The entire sentence could be removed for being obvious.

Reviewer: yeah, I don't know why the original text specified that

>       apply when the element is in that state.

Reviewer: ```suggestion
```

>     * All provided states must be active for the style to apply

Reviewer: ```suggestion
    * All provided states must be active for the style to apply.
```

          (default: infinite).

### `style[<name>;<prop1>;<prop2>;...]`
### `style[<name>:<state>;<prop1>;<prop2>;...]`

* Set the style for the named element `name`.

> * <state> is a comma-separated list of states.

Reviewer 2: `<state>` is interpreted as HTML. Please use \`state\` like above.
Also in below in `style_type`

	* If a state that the element supports is provided, the style will only
	  apply when the element is in that state.

> 	* All provided states must be active for the style to apply

Reviewer 2: Please use spaces in the docs.

Formspec elements can be themed using the style elements:

    style[<name 1>,<name 2>;<prop1>;<prop2>;...]
    style[<name 1>:<state>,<name 2>:<state>;<prop1>;<prop2>;...]
    style_type[<type 1>,<type 2>;<prop1>;<prop2>;...]
    style_type[<type 1>:<state>,<type 2>:<state>;<prop1>;<prop2>;...]

Where a prop is:

> 

Reviewer: maybe also make a note of state somewhere around here

    * noclip - boolean, set to true to allow the element to exceed formspec bounds.
    * textcolor - color. Default white.

### Valid States

> * <all elements>

Reviewer 2: Also interpreted as HTML. How about italics?
```
* *all elements*
```
